api/server/server.go

```diff
@@ -43,15 +43,22 @@ type Server struct {
 	Datastore    models.Datastore
 }
 
-func New(ctx context.Context, ds models.Datastore, mq models.MessageQueue, r *runner.Runner, tasks chan task.Request, enqueue models.Enqueue) *Server {
+type Components struct {
```

Comment:
(Continuing from above).

One package/struct that bind all together, and use this package as an extension point for customization.